src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -69,31 +78,74 @@ class PeerActor(nodeInfo: NodeInfo) extends Actor with ActorLogging {
+    handlePingMsg(rlpxConnection) orElse {
+    case MessageReceived(status: Status) =>
+      timeout.cancel()
+      log.info("Peer returned status ({})", status)
+      rlpxConnection ! RLPxConnectionHandler.SendMessage(
+        GetBlockHeaders(Left(DaoBlockNumber), maxHeaders = 1, skip = 0, reverse = 0)
+      )
+      val waitingForDaoTimeout = system.scheduler.scheduleOnce(waitForChainCheck, self, DaoHeaderReceiveTimeout)
+      context become waitingForChainForkCheck(rlpxConnection, status, waitingForDaoTimeout)
+    case StatusReceiveTimeout =>
+      log.warning("Timeout while waiting status")
+      disconnectFromPeer(rlpxConnection, Disconnect.Reasons.TimeoutOnReceivingAMessage)
+  }
 
+  def waitingForChainForkCheck(rlpxConnection: ActorRef, status: Status, timeout: Cancellable): Receive = handleTerminated orElse
+    handleDisconnectMsg orElse
+    handlePingMsg(rlpxConnection) orElse {
+    case MessageReceived(msg@BlockHeaders(blockHeader +: Nil)) if blockHeader.number == DaoBlockNumber =>
+      timeout.cancel()
+      log.info("DAO Fork header received from peer - {}", Hex.toHexString(blockHeader.hash))
+      if (daoForkValidator.validate(msg).isEmpty) {
+        log.warning("Peer is running the ETC chain")
         context become new HandshakedHandler(rlpxConnection).receive
       } else {
-        log.info("Peer does not support eth {} protocol. Disconnecting.", Message.PV63.toByte)
-        rlpxConnection ! SendMessage(Disconnect(Disconnect.Reasons.IncompatibleP2pProtocolVersion))
-        context.system.scheduler.scheduleOnce(5.seconds, self, PoisonPill)
+        log.warning("Peer is not running the ETC fork, disconnecting")
+        disconnectFromPeer(rlpxConnection, Disconnect.Reasons.UselessPeer)
+      }
+    case MessageReceived(BlockHeaders(Nil)) =>
+      // FIXME We need to do some checking related to our blockchain. If we haven't arrived to the DAO block we might
+      // take advantage of this peer and grab as much blocks as we can until DAO.
+      // ATM we will only check by DaoBlockTotalDifficulty
+      if (status.totalDifficulty < DaoBlockTotalDifficulty) {
+        log.info("Peer probably hasn't arrived to DAO yet")
+        context become new HandshakedHandler(rlpxConnection).receive
       }
+    case DaoHeaderReceiveTimeout => disconnectFromPeer(rlpxConnection, Disconnect.Reasons.TimeoutOnReceivingAMessage)
+  }
+
+  private def disconnectFromPeer(rlpxConnection: ActorRef, reason: Int): Unit = {
+    rlpxConnection ! RLPxConnectionHandler.SendMessage(Disconnect(Disconnect.Reasons.TimeoutOnReceivingAMessage))
+    context.system.scheduler.scheduleOnce(5.seconds, self, PoisonPill)
+  }
+
+  // FIXME This is a temporal function until we start keeping our status
+  private def ourStatus: Status = {
+    val totalDifficulty = 34351349760L
+    Status(
+      protocolVersion = Message.PV63, 1, totalDifficulty,
+      ByteString(Hex.decode("88e96d4537bea4d9c05d12549907b32561d3bf31f45aae734cdc119f13406cb6")),
```

Comment:
minor:
if storage is empty should this be set to genesis block hash instead of first block hash?